MAGN-6269 Move resources from DynamoIcons project to each project's resources

aosyatnik opened the pull request:
### Purpose

Delete DynamoIcons project. Add RESX files from this project in appropriate projects.

Link to YouTrack:
[MAGN-6269](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6269) Move resources from DynamoIcons project to each project's resources

### Declarations

- [x] The code base is in a better state after this PR
- [x] The level of testing this PR includes is appropriate

### Reviewers

@Benglin

sharadkjaiswal commented on src/DynamoCore/DynamoCore.csproj:
```diff
@@ -350,11 +352,15 @@ xcopy /Y "$(SolutionDir)..\extern\LibG_221\*" "$(OutputPath)libg_221\"
+	<GenerateResource UseSourcePath="true" Sources="$(ProjectDir)ProtoGeometryImages.resx" OutputResources="$(ProjectDir)ProtoGeometryImages.resources" References="$(FrameworkAssembliesPath)System.Drawing.dll" />
+    <AL TargetType="library" EmbedResources="$(ProjectDir)ProtoGeometryImages.resources" OutputAssembly="$(OutDir)ProtoGeometry.customization.dll" />
+    <GenerateResource UseSourcePath="true" Sources="$(ProjectDir)DynamoCoreImages.resx" OutputResources="$(ProjectDir)DynamoCoreImages.resources" References="$(FrameworkAssembliesPath)System.Drawing.dll" />
+    <AL TargetType="library" EmbedResources="$(ProjectDir)DynamoCoreImages.resources" OutputAssembly="$(OutDir)DynamoCore.customization.dll" />
```
Can you please also ensure that the output file gets the version information inherited from project. Currently all *.customization.dll shows file version as 0.0.0.0

Benglin commented on src/DynamoIcons/DynamoIcons.csproj:
```diff
@@ -1,82 +0,0 @@
-    <Reference Include="System.Drawing" />
-  </ItemGroup>
-  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <Target Name="BeforeBuild" Condition=" '$(OS)' != 'Unix' ">
```
This condition is removed from the other changes below, which means we no longer check to see if it's built for `Windows` before generating these `icon.customization.dll` files. @pboyer can correct me if I'm wrong.

Benglin commented on src/DynamoIcons/DynamoIcons.csproj:
```diff
@@ -1,82 +0,0 @@
-  <Target Name="BeforeBuild" Condition=" '$(OS)' != 'Unix' ">
-    <GetReferenceAssemblyPaths TargetFrameworkMoniker=".NETFramework, Version=v2.0">
-      <Output TaskParameter="FullFrameworkReferenceAssemblyPaths" PropertyName="FrameworkAssembliesPath" />
-    </GetReferenceAssemblyPaths>
```
The `Version=v2.0` caused me to have to install `.NET Framework 2.0`, can this be made `v4.0` like other projects in the solution?

aosyatnik replied:
What for should I do it? It's deleted file `DynamoIcons.csproj`. The whole DynamoIcons should be deleted.

aosyatnik commented:
Hi, @Benglin , I've changed Framework version to 4.0. Also I've added check for current platform (e.g. Windows or Unix).
Also, it's written, that this PR has merge conflicts, but it's up-to-date with master...

Benglin commented on src/DynamoCore/DynamoCore.csproj:
```diff
@@ -293,6 +294,7 @@ limitations under the License.
       <LastGenOutput>Resources.en-US.Designer.cs</LastGenOutput>
       <SubType>Designer</SubType>
     </EmbeddedResource>
+    <EmbeddedResource Include="ProtoGeometryImages.resx" />
```
Sorry for not catching this earlier, @aosyatnik. `DynamoCore` was designed without geometry in mind, so it cannot know anything about geometries. Having this here in the project works, but it is incorrect. For now I think the best place to move this is to the `Libraries\GeometryUiWpf` project. Sorry for the late request!

Benglin replied:
One more thing, please make sure the `ProtoGeometry.customization.dll` is in the right folder (not inside the `nodes` folder because that is where `GeometryUiWpf.dll` is generated).

RodRecker commented:
@aosyatnik Please use the new pull request template now.

Benglin commented:
Oops, my bad @RodRecker, @aosyatnik wasn't in the original communication and I forgot to point that out. Here is the list of all templates you can use, Anna:

https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template

Benglin commented:
Also, @aosyatnik, please be sure to update the `.gitignore` to exclude the following files:

	src/DynamoCore/DynamoCoreImages.resources
	src/DynamoCore/ProtoGeometryImages.resources
	src/Libraries/Analysis/AnalysisImages.resources
	src/Libraries/CoreNodeModels/DSCoreNodesUIImages.resources
	src/Libraries/CoreNodes/DSCoreNodesImages.resources
	src/Libraries/DSOffice/DSOfficeImages.resources
	src/Libraries/DynamoUnits/DynamoUnitsImages.resources
	src/Libraries/DynamoWatch3D/DynamoWatch3DImages.resources
	src/Libraries/IronPython/DSIronPythonNodeImages.resources
	src/Libraries/Tesellation/TessellationImages.resources
	src/Libraries/UnitsUI/UnitsUIImages.resources

Benglin commented:
Thanks @aosyatnik, it builds fine and works great on my machine now. LGTM!
